src/components/modals/ModalController.js

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { Box } from '@makerdao/ui-components-core';
+import EsmStakeModal from '../EsmStakeModal';
+import templates from './templates';
+
+const modals = {
+  esmstake: ({ onClose }) => (
```

Comment:
The part of the current modal pattern that bugs me the most is not so much having a hook and a provider, but this kind of centralization of unrelated components in one parent just because they happen to be modals. This could be avoided by passing a render function instead of just a string like "esmstake".